dotnet/src/Experimental/Process.Runtime.Dapr/Actors/ProcessActor.cs

```diff
@@ -383,10 +412,28 @@ private async Task<DaprProcessInfo> ToDaprProcessInfoAsync()
     /// Scopes the Id of a step within the process to the process.
     /// </summary>
     /// <param name="actorId">The actor Id to scope.</param>
+    /// <param name="scopeToParent">Indicates if the Id should be scoped to the parent process.</param>
     /// <returns>A new <see cref="ActorId"/> which is scoped to the process.</returns>
-    private ActorId ScopedActorId(ActorId actorId)
+    private ActorId ScopedActorId(ActorId actorId, bool scopeToParent = false)
+    {
+        if (scopeToParent && string.IsNullOrWhiteSpace(this.ParentProcessId))
+        {
+            throw new InvalidOperationException("The parent process Id must be set before scoping to the parent process.");
+        }
+
+        string id = scopeToParent ? this.ParentProcessId! : this.Id.GetId();
+        return new ActorId($"{id}.{actorId.GetId()}");
+    }
+
+    /// <summary>
+    /// Generates a scoped event for the step.
+    /// </summary>
+    /// <param name="daprEvent">The event.</param>
+    /// <returns>A <see cref="ProcessEvent"/> with the correctly scoped namespace.</returns>
+    private ProcessEvent ScopedEvent(ProcessEvent daprEvent)
     {
-        return new ActorId($"{this.Id}.{actorId.GetId()}");
+        Verify.NotNull(daprEvent);
+        return daprEvent with { Namespace = $"{this.Name}_{this._process!.State.Id}" };
```

Comment:
should there be common `namespace name generator` to ensure the namespace name remains consistent? 
I see this string format is repeated in other files too

Reply:
This one is actually different, the scoping is a bit different between the step and the (nested) process. But I have it on my backlog to refactor the scoping of things to be clearer and less error prone.